api/objects.proto

```diff
@@ -267,3 +266,62 @@ message Secret {
 	// Whether the secret is an internal secret (not set by a user) or not.
 	bool internal = 4;
 }
+
+// Config represents a set of configuration files that should be passed to a
```

Comment:
Non-blocking: the `ConfigSpec` below contains a todo about whether this should be a single file or accept multiple files as a tar.  If we settle on single file, this comment should probably be updated to reflect that (as opposed to a "set of configuration files")